api-client.ts

```diff
@@ -27,23 +29,44 @@ export class CortexApiClient {
   }
 
   async POSTForm(path: string, form: FormData) {
+    const requestSize = CortexApiClient.getFormDataSize(form);
+    if (requestSize > this.maxRequestSize) {
+      throw new Error("Request body too large");
+    }
+
     return fetch(`${this.apiUrl}/org/${this.org}${path}`, {
       method: "POST",
       headers: {
         Authorization: `Bearer ${this.accessToken}`,
       },
       body: form,
     });
+    form.values.length;
   }
 
   private async makeRequest(method: Method, path: string, body?: any) {
+    const requestBody = body ? JSON.stringify(body) : undefined;
+    // Note that we use character size instead of byte size. This is still a useful heuristic as we don't want to incur the overhead
+    // of using TextEncoder to calculate the precise byte count
+    if (requestBody && requestBody.length >= this.maxRequestSize) {
+      throw new Error("Request body too large");
+    }
+
     return fetch(`${this.apiUrl}/org/${this.org}${path}`, {
       method,
       headers: {
         Authorization: `Bearer ${this.accessToken}`,
         "Content-Type": "application/json",
       },
-      body: body ? JSON.stringify(body) : undefined,
+      body: requestBody,
     });
   }
+
+  private static getFormDataSize(formData: FormData) {
+    return [...formData].reduce(
+      (size, [_, value]) =>
+        size + (typeof value === "string" ? value.length : value.size),
```

Comment:
If `value` is not a string, will it always have a `size`? I think `size + undefined === NaN`

Reply:
I think value can only be a string or a Blob (which has size), but I can add a guard `value.size ?? 0`?

Reply:
Yeah that could work, I'm just not familiar with the types, so wasn't sure if there might be an `any` slipped in there sometime.

Reply:
Actually, this is pretty standard and documented either as a `string` or a `Blob`, and I really don't like to be overly defensive here. If it blows up, we'll hear about it presumably